test/UnitTests/MSTest.CoreAdapter.Unit.Tests/Discovery/UnitTestDiscovererTests.cs

```diff
@@ -196,6 +201,29 @@ public void SendTestCasesShouldSendTestCasesWithNaigationDataWhenDeclaredClassFu
         }
 
         [TestMethodV1]
+        public void SendTestCasesShouldSendTestCasesWithNaigationDataWhenDeclaredAssemblyNameIsNonNull()
+        {
+            var source = "DummyAssembly.dll";
+
+            // Setup mocks.
+            this.testablePlatformServiceProvider.MockFileOperations.Setup(fo => fo.CreateNavigationSession(source))
+                .Returns((object)null);
+
+            var test = new UnitTestElement(new TestMethod("M", "C", "A", false));
+            test.TestMethod.DeclaringAssemblyName = "DA";
+
+            var testElements = new List<UnitTestElement> { test };
+
+            this.SetupNavigation(source, test, test.TestMethod.DeclaringClassFullName, test.TestMethod.Name);
+
+            // Act
+            this.unitTestDiscoverer.SendTestCases(source, testElements, this.mockTestCaseDiscoverySink.Object);
+
+            // Assert
+            this.mockTestCaseDiscoverySink.Verify(ds => ds.SendTestCase(It.Is<TestCase>(tc => VerifyTestCase(tc))), Times.Once);
```

Comment:
Updated test is named `SendTestCasesShouldUseNaigationSessionForDeclaredAssemblyName` which I think is the only `SendTestCases` responsibility that needs to be tested at this layer.